### PR TITLE
Use custom domain for Edge endpoint when configured

### PR DESCRIPTION
**Pull request by kevinlind**
## Description
Read Configuration shared state for 'edge.domain' key. Change EdgeEndpoint from enum to struct to hold endpoint URL as String. Build URL based on type (prod, pre-prod, int) and custom domain.

Internal ticket: MOB-14959

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by emdobrin — COMMENTED** [review]
Looks good overall 👍 , left few comments below

File: Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift [hunk 1]
```diff
@@ -182,7 +182,8 @@ class EdgeHitProcessor: HitProcessing {
         }
 
         let edgeEndpointStr = configSharedState[EdgeConstants.SharedState.Configuration.EDGE_ENVIRONMENT] as? String
-        let edgeEndpoint = EdgeEndpoint(optionalRawValue: edgeEndpointStr?.lowercased())
+        let edgeDomainStr = configSharedState[EdgeConstants.SharedState.Configuration.EDGE_DOMAIN] as? String
+        let edgeEndpoint = EdgeEndpoint(type: EdgeEndpointType(optionalRawValue: edgeEndpointStr?.lowercased()), optionalDomain: edgeDomainStr)
```
[the remark this answers is not in the document]
Comment (kevinlind) [reply Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift]: I'm not clear on this comment. Can you give me an example?  Where I'm calling `lowercased` seems like a good spot.

[the remark this answers is not in the document]
Comment (emdobrin) [reply Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift]: I was thinking we can move it in EdgeEndpointType for consistency, similar with how we handle the domain internally.

File: Sources/EdgeNetworkHandlers/EdgeEndpoint.swift [hunk 2]
```diff
@@ -13,7 +13,7 @@
 import Foundation
 
 /// Represents all the known endpoints for the Edge Network
```
Comment (emdobrin) [inline Sources/EdgeNetworkHandlers/EdgeEndpoint.swift:15]: Let's update the comments on this enum since is not directly related to the endpoints. E.g. Represents all the known Edge Network environment types

**Review by emdobrin — APPROVED** [review]
👍
